src/entitysdk/client.py

```diff
@@ -349,8 +350,9 @@ def download_directory(
         output_path: os.PathLike,
         project_context: ProjectContext | None = None,
         ignore_directory_name: bool = False,
+        max_concurrent: int = 4,
```

Comment:
Should we make it serial by default to avoid surprises?

Reply:
I was thinking of having `max_concurrent` not use threads at all, in case it becomes a problem, but I don't forsee it being one (famous last words.)
I think doing it in parallel by default is worthwhile, personally.

Reply:
Just to be aware, if we integrate this in a service using fastapi (obi-one), and we have N concurrent requests served by sync endpoints, then we'll create up to `N*max_concurrent` additional threads (besides the N needed by the fastapi pool).

Reply:
do you think that's a problem?
obi-one can limit the max_concurrent, if it wants

Reply:
Yeah, but bringing a service to a halt/crash in order to realize that max_concurrent has to be lowered in download_directory is not great, is it?

Reply:
I would be surprised if threads doing downloads is what pushes it over the edge; assuming all 40 threads are spawning another 4, that's a limit of 160 - which is high, but not unheard of.

Reply:
Ok, I switched the default to 1